src/webui/webapplication.h

```diff
@@ -29,92 +29,101 @@
+class WebApplication final : public AbstractWebApplication
 {
+    Q_OBJECT
     Q_DISABLE_COPY(WebApplication)
 
+#ifndef Q_MOC_RUN
+#define PUBLIC_API
+#endif
+
 public:
     explicit WebApplication(QObject *parent = nullptr);
 
-private:
-    // Actions
-    void action_public_webui();
-    void action_public_index();
-    void action_public_login();
-    void action_public_logout();
-    void action_public_theme();
-    void action_public_images();
-    void action_query_torrents();
-    void action_query_preferences();
-    void action_query_transferInfo();
-    void action_query_propertiesGeneral();
-    void action_query_propertiesTrackers();
-    void action_query_propertiesWebSeeds();
-    void action_query_propertiesFiles();
-    void action_query_getLog();
-    void action_query_getPeerLog();
-    void action_query_getPieceHashes();
-    void action_query_getPieceStates();
-    void action_sync_maindata();
-    void action_sync_torrent_peers();
-    void action_command_shutdown();
-    void action_command_download();
-    void action_command_upload();
-    void action_command_addTrackers();
-    void action_command_resumeAll();
-    void action_command_pauseAll();
-    void action_command_resume();
-    void action_command_pause();
-    void action_command_setPreferences();
-    void action_command_setFilePrio();
-    void action_command_getGlobalUpLimit();
-    void action_command_getGlobalDlLimit();
-    void action_command_setGlobalUpLimit();
-    void action_command_setGlobalDlLimit();
-    void action_command_getTorrentsUpLimit();
-    void action_command_getTorrentsDlLimit();
-    void action_command_setTorrentsUpLimit();
-    void action_command_setTorrentsDlLimit();
-    void action_command_alternativeSpeedLimitsEnabled();
-    void action_command_toggleAlternativeSpeedLimits();
-    void action_command_toggleSequentialDownload();
-    void action_command_toggleFirstLastPiecePrio();
-    void action_command_setSuperSeeding();
-    void action_command_setForceStart();
-    void action_command_delete();
-    void action_command_deletePerm();
-    void action_command_increasePrio();
-    void action_command_decreasePrio();
-    void action_command_topPrio();
-    void action_command_bottomPrio();
-    void action_command_setLocation();
-    void action_command_rename();
-    void action_command_setAutoTMM();
-    void action_command_recheck();
-    void action_command_setCategory();
-    void action_command_addCategory();
-    void action_command_removeCategories();
-    void action_command_getSavePath();
-    void action_version_api();
-    void action_version_api_min();
-    void action_version_qbittorrent();
-
-    typedef void (WebApplication::*Action)();
-
-    QString scope_;
-    QString action_;
-    QStringList args_;
+private slots:
+    // API Methods
+    PUBLIC_API void login();
+    PUBLIC_API void getVersion();
+    PUBLIC_API void getAppVersion();
+    PUBLIC_API void getThemeItem();
+    void logout();
```

Comment:
Is it possible to declare all these as const methods?
e.g. `void logout() const;`

Reply:
Why not? I'll try to do it later.

Reply:
Oops... They call non-const base members.